### DISCUSSION TITLE
What's the difference between these two elements?

### DISCUSSION THREAD

**Discussion by Roky3029** [theOehrly/Fast-F1#504] [Q&A]
Hi, I've got one of the rows from `fastf1.get_events_remaining()` and I was wondering what was the difference between, for example, `Session5DateUtc` and `Session5Date`. I'm asking this because the first one gives me a date in the Epoch format, and the second one returns `null`.
In addition to my first question, why is it that `Session5Date` returns null? And how does `Session5DateUtc` return a specific time although in the official page of F1 it's still TBC?

EDIT: After converting the data to JSON, and fetching it with TypeScript, this is how it looks like:
```js
{
  RoundNumber: 1,
  Country: 'Bahrain',
  Location: 'Sakhir',
  OfficialEventName: '',
  EventDate: 1709337600000,
  EventName: 'Bahrain Grand Prix',
  EventFormat: 'conventional',
  Session1: 'Practice 1',
  Session1Date: null,
  Session1DateUtc: 1709164800000,
  Session2: 'Practice 2',
  Session2Date: null,
  Session2DateUtc: 1709164800000,
  Session3: 'Practice 3',
  Session3Date: null,
  Session3DateUtc: 1709251200000,
  Session4: 'Qualifying',
  Session4Date: null,
  Session4DateUtc: 1709251200000,
  Session5: 'Race',
  Session5Date: null,
  Session5DateUtc: 1709337600000,
  F1ApiSupport: true
}
```

Python file: 
```py
...
@next_event.route("/next_event")
def next_event():
    next_event = fastf1.get_events_remaining()

    return next_event.iloc[0].to_json()
...
```

TypeScript Fetching
```ts
...
export const getNextRace = async () => {
	const url = `${SERVER_LINK}/next_event`
	const nextRace = await fetch(url)
	const nextRaceData = (await nextRace.json()) as NextEvent

	return nextRaceData
}
```

**Comment by Casper-Guo** [comment]
`Date` usually refers to local time within FastF1 while `DateUTC` is UTC.

Can you edit your comment to show the full output please?

**Reply by Roky3029** [reply]
It's already updated :)

**Comment by theOehrly** [comment]
> I was wondering what was the difference between, for example, `Session5DateUtc` and `Session5Date`. 

As @Casper-Guo stated, one is UTC, one is timezone-aware local time

https://docs.fastf1.dev/events.html#event-schedule-data

> the first one gives me a date in the Epoch format

that has to be due to you serializing the data to JSON

> and the second one returns `null`

There are three backends for the schedule data. My own backend, F1's timing data API and Ergast. Currently, only Ergast has a (preliminary) schedule for the 2024 season. Therefore, FastF1 falls back to using Ergast. Ergast does provide UTC timestamps and no timezone information, therefore when this backend is used, only the UTC Dates can be provided. The other two backends support the additional information.
(I just noticed that the info around Ergast and timezone info is written somewhat ambiguously in the documentation. I'll fix that)


> And how does `Session5DateUtc` return a specific time although in the official page of F1 it's still TBC?

![grafik](https://github.com/theOehrly/Fast-F1/assets/23384863/0a88e9d7-e484-4ce1-89c4-662bbd714711)

It returns 00:00 as time, same as all the other timestamps. Yes, technically that's a specific time. But so is the unix timestamp of the same value. I can't just easily remove the time here.

**Reply by Roky3029** [reply]
So I guess that when dates are confirmed, along with the exact hour, it will automatically updated and thus the exact hour showed when fetching the API, am I right?

**Reply by theOehrly** [reply]
Yes, correct. There might be some delay between the confirmation of the exact time and that information being available through one of the APIs though.
